src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,51 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  get isAlive() {
+    return this.health > 0;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health <= 0) {
+      Animal.cleanupDeadAnimals();
+    }
+  }
 }
 
+Animal.alive = [];
+Animal.cleanupDeadAnimals = () => {
+  Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+};
```

Comment:
The current implementation of `cleanupDeadAnimals` uses a filter inside a class method, which is good for performance as it avoids nesting loops. However, be cautious about where this method is called. It's currently being called every time an animal's health reaches 0 or below, which could lead to performance issues if many animals are taking damage simultaneously. Consider invoking this cleanup method less frequently or in a batch process to ensure better performance. [CODE PERFORMANCE]